### DISCUSSION TITLE
Math Tagging Examples

### DISCUSSION THREAD

**Discussion by davidcarlisle** [latex3/tagging-project#771] [General]
The collection of [WTPDF / PDF/UA-2 Examples](https://github.com/latex3/tagging-project/discussions/72) has been reprocessed with the latest LaTeX code and the [PDFs folder on google drive](https://drive.google.com/drive/folders/17iGeBcboFMx410H1Rozhb-Io-cp0f3TB?usp=sharing) updated.

Here we highlight the examples showing Math tagging.

Most of the files are in three versions:

Math Examples are shown in three forms

 * _file-tagged_.pdf (or in some cases just _file_.pdf)

   This tagging uses an externally provided MathML conversion and tags each Formula with Alt (containing the LaTeX source), an Associated file with TeX source and an associated File with the supplied MathML.

  * _file_-luamml.pdf

    This version tags each formula with MathML directly generated by LaTeX (using the new `luamml` package).

  * _file_-se.pdf

    This version uses MathML namespace Structure Elements generated by LaTeX (using the `luamml` package) rather than an associated file.

Two small examples are available that show most of the features.

## example-1

A basic example, but note the Square Root which requires the √ character to be marked as an artifact
when using  Structure Elements to avoid it being doubled as it is also generated by the `/mroot` element.

[TeX source (github)](https://github.com/latex3/tagging-project/tree/main/project-examples/example-1)

Supplied MathML AF: [PDF](https://drive.google.com/file/d/1wRbsE8w6cATyEBJtvLu0LLdCg-CAhQc0/view?usp=drive_link),  [ngPDF](https://ngpdf.com/pdfUrlToHtml?mathjax=false&prettyprint=true&mode=editor&pdfUrl=https://drive.usercontent.google.com/download?id=1wRbsE8w6cATyEBJtvLu0LLdCg-CAhQc0&export=download&authuser=0)

luamml MathML AF: [PDF](https://drive.google.com/file/d/1aPzFVPcIT-1bbFxW96nVZRYrQkPEAHw5/view?usp=drive_link), [ngPDF](https://ngpdf.com/pdfUrlToHtml?mathjax=false&prettyprint=true&mode=editor&pdfUrl=https://drive.usercontent.google.com/download?id=1aPzFVPcIT-1bbFxW96nVZRYrQkPEAHw5&export=download&authuser=0)

luamml MathML Structure Element: [PDF](https://drive.google.com/file/d/1f0jQs8CUXDT6F5s8PKui3CP5BeWfuyCY/view?usp=drive_link), [ngPDF](https://ngpdf.com/pdfUrlToHtml?mathjax=false&prettyprint=true&mode=editor&pdfUrl=https://drive.usercontent.google.com/download?id=1f0jQs8CUXDT6F5s8PKui3CP5BeWfuyCY&export=download&authuser=0)

## nested-1

Example showing some more complicated uses with nested math constructs and linking.

[TeX source (github)](https://github.com/latex3/tagging-project/tree/main/project-examples/nested1)

Supplied MathML AF: [PDF](https://drive.google.com/file/d/1EjKvvJh1N2wn62027AVqw4ZPWiGBDczR/view?usp=drive_link), [ngPDF](https://ngpdf.com/pdfUrlToHtml?mathjax=false&prettyprint=true&mode=editor&pdfUrl=https://drive.usercontent.google.com/download?id=1EjKvvJh1N2wn62027AVqw4ZPWiGBDczR&export=download&authuser=0)

luamml MathML AF: [PDF](https://drive.google.com/file/d/1R1ZOoFdt1s3N71HET6A8HcMuyZLrQdiM/view?usp=drive_link), [ngPDF](https://ngpdf.com/pdfUrlToHtml?mathjax=false&prettyprint=true&mode=editor&pdfUrl=https://drive.usercontent.google.com/download?id=1R1ZOoFdt1s3N71HET6A8HcMuyZLrQdiM&export=download&authuser=0)

luamml MathML Structure Element: [PDF](https://drive.google.com/file/d/12ML7dIvX9KUN7ha67RngY0k1wwG9GlYO/view?usp=drive_link), [ngPDF](https://ngpdf.com/pdfUrlToHtml?mathjax=false&prettyprint=true&mode=editor&pdfUrl=https://drive.usercontent.google.com/download?id=12ML7dIvX9KUN7ha67RngY0k1wwG9GlYO&export=download&authuser=0)

----

Several more (bigger) examples, are available in the  [PDFs folder on google drive](https://drive.google.com/drive/folders/17iGeBcboFMx410H1Rozhb-Io-cp0f3TB?usp=sharing).

----

You can use [ngPDF](https://ngpdf.com) to see the tagging structure and derivation to HTML.

The tag structure is also shown by some PDF readers, notably Acrobat Pro and PDF-XChange.

With a Beta of NVDA including the MathCat Math plugin, these documents may be read

- one can use Acrobat (Pro or Reader) to read the files using MathML structure elements,

- one can use Foxit reader to read the examples using MathML Associated Files.

We expect the new NVDA to become available early next year and by then
or soon after hopefully more PDF readers will support both types of
MathML tagging within PDF as mandated by the PDF 2.0 standard.

**Comment by car222222** [comment]
Many thanks!  A very useful post, but still (naturally) with very few examples.

When we have a good spec (or even any suitably specific information!) covering what further types or instances of examples need to be added here, then people will be able (reasonably easily) to use these files as examples and guides for how to construct further such example files, as required.
